### PR TITLE
Revert the upper-case PROJECT_NAME nonsense that was part of #2402

### PR DESCRIPTION
**Pull request by sfan5**

### CODE REVIEW

**Comment by ShadowNinja** [comment]
* This basically just renames `PROJECT_NAME_LOWER` to `PROJECT_NAME` and `PROJECT_NAME` to `PROJECT_NAME_CAPITALIZED`, except that the required code is more complex and adds more work for forks that use MixedCase.
  * This doesn't work in Travis's version of CMake.
  * I have yet to hear a good reason for this.  I discussed this with you for a while on IRC and all I understood was "we shouldn't add features (the ability for forks to easily use MixedCase) unless a lot of people want them." (and should remove them if they pop up)

**Comment by sfan5** [comment]
> * This basically just renames PROJECT_NAME_LOWER to PROJECT_NAME and PROJECT_NAME to PROJECT_NAME_CAPITALIZED, except that the required code is more complex and adds more work for forks that use MixedCase.

29 code changes where lowercase project name needed to be used
vs.
16 places that needed the capitalized project name
If you had done it like this before it would have been less code-changes
> * This doesn't work in Travis's version of CMake.
I'll fix that.
> I have yet to hear a good reason for this.
The good reason is that it's stupid to add feature to accomodate for things someone might possibly do in a fork in the future.

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,8 +6,13 @@ if(${CMAKE_VERSION} STREQUAL "2.8.2")
...
 # This can be read from ${PROJECT_NAME} after project() is called
-project(Minetest)
-string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
+project(minetest)
+
+# Put capitalized PROJECT_NAME into PROJECT_NAME_CAPITALIZED
+string(SUBSTRING ${PROJECT_NAME} 0 1 TMP1)
+string(TOUPPER ${TMP1} TMP1)
+string(SUBSTRING ${PROJECT_NAME} 1 -1 TMP2)
+string(CONCAT PROJECT_NAME_CAPITALIZED ${TMP1} ${TMP2})
```
Comment (PilzAdam) [inline CMakeLists.txt]: This is stupid. Just `set(PROJECT_NAME_CAPITALIZED "Minetest")`. Forks can change both versions if they want to.

**Comment by PilzAdam** [comment]
I'm with @ShadowNinja on this one. Setting the proper project name and having a "special" variable for the lowercase version seems more reasonable.

**Comment by kwolekr** [comment]
Guys, vote here on what to do with this PR:  http://strawpoll.me/4215136

**Comment by ShadowNinja** [comment]
> 29 lowercase lines vs. 16 capitalized lines.
> If you had done it like this before it would have been less code-changes.

But I didn't do it like that before, so it's too late to get any advantage from having a smaller patch, and this results in changing every one of those lines

>The good reason is that it's stupid to add feature to accommodate for things someone might possibly do in a fork in the future.

It's also stupid to remove a feature solely because nobody is using it currently.

File: src/cmake_config.h.in [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #define CMAKE_CONFIG_H
 
 #define PROJECT_NAME "@PROJECT_NAME@"
-#define PROJECT_NAME_LOWER "@PROJECT_NAME_LOWER@"
+#define PROJECT_NAME_C "@PROJECT_NAME_CAPITALIZED@"
```
Comment (ShadowNinja) [inline src/cmake_config.h.in]: `_C` suggests that it's a C string (`const char *`), and implies that `PROJECT_NAME` is a C++ string (`std::string`).

Comment (sfan5) [inline src/cmake_config.h.in]: Not really, no.

**Comment by sfan5** [comment]
>> 29 lowercase lines vs. 16 capitalized lines.
>> If you had done it like this before it would have been less code-changes.
>
> But I didn't do it like that before, so it's too late to get any advantage from having a smaller patch, and > this results in changing every one of those lines

Actually the patch does not touch _every_ line.
>>The good reason is that it's stupid to add feature to accommodate for things someone might possibly do in a fork in the future.
>
> It's also stupid to remove a feature solely because nobody is using it currently.

That's correct, this feature however is pretty useless because nobody is likely to use it soon.

**Comment by kwolekr** [comment]
I change my vote to "for".
It seems quite silly to optimize for the uncommon case...

**Comment by sfan5** [comment]
558d715bf62b000c9b10b44a0e3591ed4e669fa5 4ea5a96fffb1a0300f73e97d4c85bb5c32e3786d

**Pull request closed without merging**